File: Problems/509_Fibonacci_Number.cpp
class Solution {
public:
    int fib(int n) {
        int pre = 1;
        int prepre = 0;
        if(n==0) return prepre;
        if(n==1) return pre;
        int curr;
        for(int i=0; i<n-1; i++){
            curr = pre+prepre;
            prepre = pre;
            pre = curr;
        }
        return curr;
    }
};

File: Problems/509_Fibonacci_Number.py
class Solution:
    def fib(self, n: int) -> int:
        pre = 1
        prepre = 0
        if n==prepre:
            return 0
        if n==pre:
            return 1
        cnt = 1
        for _ in range(n-1):
            curr = pre+prepre
            # print(curr, pre, prepre)
            prepre = pre
            pre = curr
        return curr